materialize-mongodb/config.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+
+  "net/url"
+)
+
+type config struct {
+	Address  string `json:"address" jsonschema:"title=Address,description=Host and port of the database." jsonschema_extras:"order=0"`
+	User     string `json:"user" jsonschema:"title=User,description=Database user to connect as." jsonschema_extras:"order=1"`
+	Password string `json:"password" jsonschema:"title=Password,description=Password for the specified database user." jsonschema_extras:"secret=true,order=2"`
+	Database string `json:"database" jsonschema:"title=Database,description=Name of the database to capture from." jsonschema_extras:"order=3"`
```

Comment:
The description here `"Name of the database to capture from"` is probably not what we want.

Also: The top-level config has a database, and each resource has a database. From what I can tell, the top-level config database is only used for storing fence records, whereas each individual resource is free to specify its own database.

I believe there is some potential confusion here that needs to be resolved. My initial thought is that resources should not have their own `database` value and should instead just use the top-level one. This would arguably reduce the flexibility of resource configurations, but I think it would simplify the configuration enough to be worth the tradeoff.